greetings-tycho/2.32.0-J11/org.xtext.example.mydsl.target/org.xtext.example.mydsl.target.target

```diff
@@ -19,7 +19,7 @@
 			<repository location="https://download.eclipse.org/modeling/tmf/xtext/updates/nightly/"/>
 		</location>
 		<location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="true" type="InstallableUnit">
-			<unit id="com.google.gson" version="2.10.1.v20230109-0753"/>
+			<unit id="com.google.gson" version="2.10.1"/>
```

Comment:
Hmm the manifest of gson 2.10.1 has
```
Export-Package: com.google.gson;uses:="com.google.gson.reflect,com.googl
 e.gson.stream";version="2.10.1",
 com.google.gson.annotations;version="2.10.1",
 com.google.gson.reflect;version="2.10.1",
 com.google.gson.stream;version="2.10.1"
Import-Package: sun.misc;resolution:=optional,com.google.gson.annotations
 ```
 It imports its own package without a version constraint and exports it with a version constraint.

The orbit gson doesn't look great either:
```
Export-Package: com.google.gson.internal;x-internal:=true;version="2.10.1";uses:="com.google.gson,com.google.gson.reflect,com.google.gson.stream",
 com.google.gson.internal.bind;x-internal:=true;version="2.10.1";uses:="com.google.gson,com.google.gson.internal,com.google.gson.reflect,com
 .google.gson.stream",
 com.google.gson.internal.bind.util;x-internal:=true;version="2.10.1",
 com.google.gson.internal.reflect;x-internal:=true;version="2.10.1";uses:="com.google.gson",
 com.google.gson.internal.sql;x-internal:=true;version="2.10.1";uses:="com.google.gson,com.google.gson.i
 nternal.bind",
 com.google.gson;version="2.10.1";uses:="com.google.gson.internal,com.google.gson.reflect,com.google.gson.stream",
 com.google.gson.annotations;version="2.10.1",
 com.google.gson.reflect;version="2.10.1",
 com.google.gson.stream;version="2.10.1"
Import-Package: com.google.gson;version="[2.10,3)",
 com.google.gson.annotations;version="[2.10,3)",
 com.google.gson.internal;version="[2.10,3)",
 com.google.gson.internal.bind;version="[2.10,3)",
 com.google.gson.internal.bind.util;version="[2.10,3)",
 com.google.gson.internal.reflect;version="[2.10,3)",
 com.google.gson.internal.sql;version="[2.10,3)",
 com.google.gson.reflect;version="[2.10,3)",
 com.google.gson.stream;version="[2.10,3)",sun.misc;resolution:=optional
```

I assume the Maven definition uses substitutable API package imports of exported packages (https://docs.osgi.org/specification/osgi.core/8.0.0/framework.module.html#i3238802) so seems to be more correct.

Reply:
what a mess